DoseMathTests/DoseMathTests.swift

```diff
@@ -632,17 +712,38 @@ class RecommendBolusTests: XCTestCase {
             atDate: glucose.first!.startDate,
             lastTempBasal: nil,
             maxBolus: maxBolus,
+            maxIOB: maxIOB,
             glucoseTargetRange: glucoseTargetRange,
             insulinSensitivity: insulinSensitivitySchedule,
-            basalRateSchedule: basalRateSchedule
+            basalRateSchedule: basalRateSchedule,
+            insulinOnBoard: 0.0
         )
 
         XCTAssertEqualWithAccuracy(1.0, dose, accuracy: 1.0 / 40.0)
     }
+    
+    func testHighAndRisingAtIOB() {
+        let glucose = loadGlucoseValueFixture("recommend_temp_basal_high_and_rising")
+        
+        let dose = DoseMath.recommendBolusFromPredictedGlucose(glucose,
+                                                               atDate: glucose.first!.startDate,
+                                                               lastTempBasal: nil,
+                                                               maxBolus: maxBolus,
+                                                               maxIOB: maxIOB,
+                                                               glucoseTargetRange: glucoseTargetRange,
+                                                               insulinSensitivity: self.insulinSensitivitySchedule,
+                                                               basalRateSchedule: basalRateSchedule,
+                                                               insulinOnBoard: 20.0
+        )
+        
+        XCTAssertEqual(0.0, dose)
+    }
```

Comment:
What should happen for maxIOB = nil?  throw an error like all the other missing preferences or not have a limit?